modules/AlphaWalletFoundation/AlphaWalletFoundation/Tokens/JSON/TokenGroupIdentifier.swift

```diff
@@ -162,6 +168,10 @@ public class TokenGroupIdentifier: TokenGroupIdentifierProtocol {
     public func hasContract(address: String, chainID: Int) -> Bool {
         return decodedTokenEntries["\(address):\(chainID)"] != nil
     }
+
+    public func isSpam(address: String, chainID: Int) -> Bool {
+        return spamTokenEntries.contains("\(address):\(chainID)".lowercased())
```

Comment:
Can you extract this key generation? It's used in a few place. Looks like good as wrapped in an `enum functional {}`

Reply:
This could also be combined with the one where a "-" is used instead of ":". Like in `ContractAddressObject. generatePrimaryKey`. Maybe standardize the usage.
```swift
static func generatePrimaryKey(fromContract contract: AlphaWallet.Address, server: RPCServer) -> String {
        return "\(contract.eip55String)-\(server.chainID)"
    }
```

Reply:
This file internally first? Doesn't need to extract for the other one seems it's just a coincidence.

Reply:
but both represent the same thing